src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1687,7 +1687,7 @@ public static Tombstone TableValueToTombstone(JsonOperationContext context, ref
                 table.Delete(doc.StorageId);
 
                 if ((flags & DocumentFlags.HasAttachments) == DocumentFlags.HasAttachments)
-                    AttachmentsStorage.DeleteAttachmentsOfDocument(context, lowerId, changeVector, modifiedTicks);
+                    AttachmentsStorage.DeleteAttachmentsOfDocument(context, lowerId, changeVector, modifiedTicks, artificial: flags.Contain(DocumentFlags.Artificial));
```

Comment:
we probably want to do the same as in manipulation of other artifical docs, where mthod takes flags as an argument and you supposed to pass here: `flags: DocumentFlags.Artificial | DocumentFlags.FromResharding`

var flagsForDelete = DocumentsFlags.None;
if (flags.Contain(DocumentFlags.Artificial))
    flagsForDelete = DocumentFlags.Artificial | DocumentFlags.FromResharding;

Reply:
fixed